src/Simulator/FluidSimulator.cpp

```diff
@@ -128,100 +128,94 @@ bool cFluidSimulator::IsHigherMeta(NIBBLETYPE a_Meta1, NIBBLETYPE a_Meta2)
 
 
 
-// TODO Not working very well yet :s
-Direction cFluidSimulator::GetFlowingDirection(int a_X, int a_Y, int a_Z, bool a_Over)
+Vector3f cFluidSimulator::GetFlowingDirection(int a_X, int a_Y, int a_Z)
 {
+	Vector3f vDirection;
+	Vector3i Points[4];
+	int LevelPoint[4];
```

Comment:
Instead of declaring your variables at the top of the function, you should introduce them as they are needed and initialise them in the same statement.